nncf/openvino/quantization/helpers.py

```diff
@@ -0,0 +1,107 @@
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import tempfile
+from pathlib import Path
+from typing import Optional
+from typing import Callable
+
+import openvino.runtime as ov
+from openvino.tools import pot
+
+from nncf.data import DataLoader
+from nncf.openvino.utils import POTDataLoader
+from nncf.openvino.engine import CustomEngine
+
+
+def _convert_openvino_model_to_compressed_model(model: ov.Model,
+                                                target_device: str) -> pot.graph.nx_model.CompressedModel:
+    """
+    Serializes the provided OpenVINO model and loads the model in the POT representation.
+
+    :param model: The OpenVINO model.
+    :param target_device: The target device.
+    :return: The POT representation of the provided model.
+    """
+    with tempfile.TemporaryDirectory(dir=tempfile.gettempdir()) as tmp_dir:
+        xml_path = str(Path(tmp_dir).joinpath('model.xml'))
+        bin_path = str(Path(tmp_dir).joinpath('model.bin'))
+        ov.serialize(model, xml_path, bin_path)
+        model_config = {
+            'model_name': 'model',
+            'model': xml_path,
+            'weights': bin_path,
+        }
+        pot_model = pot.load_model(model_config, target_device)
+
+    return pot_model
+
+
+def _convert_compressed_model_to_openvino_model(model: pot.graph.nx_model.CompressedModel) -> ov.Model:
+    """
+    Saves the provided POT compressed model and loads it as `openvino.runtime.Model` object.
+
+    :param model: The POT compressed model.
+    :return: The `openvino.runtime.Model`  object which represents the provided model.
+    """
+    with tempfile.TemporaryDirectory(dir=tempfile.gettempdir()) as tmp_dir:
+        paths = pot.save_model(model, save_path=tmp_dir, model_name='model')
+        xml_path = paths[0]['model']
+        bin_path = paths[0]['weights']
+        ie = ov.Core()
+        ov_model = ie.read_model(xml_path, bin_path)
+    return ov_model
+
+
+def quantize_impl(model: ov.Model,
+                  calibration_dataset: DataLoader,
+                  preset: str,
+                  target_device: str,
+                  subset_size: int,
+                  fast_error_correction: bool,
+                  model_type: Optional[str] = None) -> ov.Model:
+    """
+    Implementation of the `quantize()` method for the OpenVINO backend.
+    """
+    pot_model = _convert_openvino_model_to_compressed_model(model, target_device)
+
+    engine_config = {
+        'device': 'CPU',
+        'stat_requests_number': 2,
```

Comment:
@alexsu52, @KodiaqQ and everybody, we need to decide how we can expose such backend-specific stuff through a common API that is being developed now.

Reply:
The ```class AdditionalQuantizationParamters``` will has the ```backend``` attribute where the user can assign instance of the ``OpenVINOParameters```, for example.  OpenVINOParameters will consist ```stat_requests_number``` attribute.